handlers/passport.js

```diff
@@ -16,54 +16,29 @@ module.exports = (passport) => {
     User.findById(id, (err, user) => { done(err, user); });
   });
 
-  passport.use(new GoogleStrategy({
-    clientID: config.auth.google.clientID,
-    clientSecret: config.auth.google.clientSecret,
-    callbackURL: config.auth.google.callbackURL,
-  }, (token, refreshToken, profile, done) => {
+  passport.use(new GoogleStrategy(auth.google, (token, refreshToken, profile, done) => {
     process.nextTick(() => {
-      User.findOne({ 'google.id': profile.id }, (err, user) => {
-        if (err) return done(err); // throw err
+      User.findOne({ 'google.id': profile.id }).then((user) => {
         if (!user) {
           const newUser = new User(); // create a new user
-          newUser.google.id = profile.id;
-          newUser.google.token = token;
-          newUser.google.name = profile.displayName;
-          newUser.google.email = (profile.emails[0].value || '').toLowerCase();
-          newUser.save((err) => {
-            if (err) throw err;
-            return done(null, newUser);
-          });
+          newUser.google = auth.google.setProfile(profile, token);
+          return newUser.save().then(doc => done(null, doc)).catch((err) => { throw err; });
```

Comment:
This it is the line that I don't get it

Reply:
setprofile is declared in the auth strategies object. It parses the providers profile into out own user objects. It could be moved into the user model i think.
Then we return the newUser object or throw an error if it fails to do so.

Reply:
Ok, and let me see if I understand the next code: 
then(doc => done(null, doc)) 
That's a Promises  that receive the new User and then is pass it to the done function and returns the call?

Reply:
yeah, `then` executes after the query is done and receives the promise object that was just created if successful, if there is an error it goes straight to the `catch`.

Reply:
Got it, is a little bit tricky at beginning.
Practicing is the best way to learn this way of programing